source/features/search-in-current-path.tsx

```diff
@@ -0,0 +1,48 @@
+import select from 'select-dom';
+import onetime from 'onetime';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {getCleanPathname} from '../github-helpers';
+
+let search = '';
+
+function getSearch(): string {
+	const path = getCleanPathname().split('/');
+
+	if (pageDetect.isRepoTree() && !pageDetect.isRepoRoot()) {
+		return `path:${path.slice(4).join('/')} `;
+	}
+
+	if (pageDetect.isSingleFile() && path.length !== 5) {
+		return `path:${path.slice(4, -1).join('/')} `;
+	}
+
+	return '';
+}
+
+function setSearch(): void {
+	const searchInput = select<HTMLInputElement>('[data-hotkey="s,/"]')!;
+
+	if (searchInput.value === search) {
+		search = getSearch();
+		searchInput.value = search;
+	}
+}
+
+function init(): void {
+	setSearch();
+	document.addEventListener('pjax:end', setSearch);
```

Comment:
We already handle this event. Avoid `onetime` and possibly use `repeatOnBackButton` instead.
`setSearch` should be the `init`